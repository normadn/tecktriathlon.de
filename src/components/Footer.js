import React from "react";
import moment from "moment";

import ContentWrapper from "./ContentWrapper";
import SignUpForm from "../components/SignUpForm";

import mail from "../assets/img/icons/socialMail.svg";
import twitter from "../assets/img/icons/socialTwitter.svg";
import logo from "../assets/img/logo/amsel-logo-bird-small-white-new.svg";
import facebook from "../assets/img/icons/socialFacebook.svg";
import instagram from "../assets/img/icons/socialInstagram.svg";
import wordmark from "../assets/img/logo/amsel-tech-word-white.svg";
import MediaQuery from "react-responsive";

const navigateLinks = [
  {
    label: "Impressum",
    url: "/impressum",
    dataCy: "footer-2018-link"
  },
  {
    label: "Datenschutz",
    url: "/impressum",
    dataCy: "footer-lhd-link"
  }
];

const socialLinks = [
  {
    icon: mail,
    url: "mailto:norman@amsel.tech",
    label: "norman@amsel.tech",
    dataCy: "footer-email-link"
  },
  {
    label: "07023 9577753",
  }
];

const finePrint = {
  color: "#bebebe",
  fontSize: "12px"
};

const [smallScreens, mediumScreens, largeScreens] = [
  "@media(max-width: 950px)",
  "@media (min-width: 950px) and (max-width: 1200px)",
  "@media(min-width: 1200px)"
];

const Footer = () => (
  <footer css={{ backgroundColor: "#11213f", color: "#ffffff" }}>
    <ContentWrapper>
      <div
        css={{
          display: "grid",
          gridTemplateColumns: "auto auto auto",
          paddingTop: "80px",
          " h3": {
            color: "inherit",
            textTransform: "uppercase"
          },
          [smallScreens]: {
            display: "block",
            textAlign: "center"
          }
        }}
      >
        <div
          css={{
            [smallScreens]: {
              marginLeft: "auto",
              marginRight: "auto"
            },
            maxWidth: "455px"
          }}
        >
          <span>
            <img
              height="34px"
              src={logo}
              css={{ marginBottom: "5px", paddingRight: "13px" }}
              alt="QHacks Crown"
            />
            <img height="37px" src={wordmark} alt="QHacks Wordmark" />
          </span>
          <p
            css={{
              fontSize: "12px",
              lineHeight: "20px",
              paddingTop: "40px",
              paddingRight: "20px",
              paddingBottom: "40px",
              lineHeight: "1.67",
              [smallScreens]: { paddingLeft: "24px", paddingRight: "24px" }
            }}
          >
            Die Software Argentur in Bissingen an der Teck.
          </p>
          
          <div css={{ [smallScreens]: { display: "none" } }}>
            <p
              data-cy="footer-address"
              css={{ ...finePrint, marginBottom: "5px" }}
            >
              Amsel.tech | Amselweg 7, 73266 Bissingen an der Teck
            </p>
            <p
              data-cy="footer-dev-plug"
              css={{ ...finePrint, fontWeight: "400" }}
            >
              {"< />"} with ♡ by Amsel-tech
            </p>
            <p
              data-cy="footer-copyright-notice"
              css={{
                ...finePrint,
                paddingBottom: "32px",
                marginTop: "6px"
              }}
            >
              Copyright © {moment().year()} Amsel.tech
            </p>
          </div>
        </div>
        <div
          css={{
            marginLeft: "auto",
            marginRight: "auto",
            paddingTop: "80px",
            [smallScreens]: {
              paddingTop: "0px",
              paddingBottom: "50px"
            }
          }}
        >
          <h3>Rechtliches</h3>
          <ul
            css={{
              paddingTop: "30px",
              listStyleType: "none",
              ">li": { paddingBottom: "23px" },
              " a": {
                color: "inherit",
                fontWeight: 600,
                textDecoration: "none",
                ":hover": {
                  color: "#ed253a"
                }
              }
            }}
          >
            {navigateLinks.map(({ label, url, dataCy }) => (
              <li key={url}>
                <a href={url} data-cy={dataCy}>{`${label} ▸`}</a>
              </li>
            ))}
          </ul>
        </div>
        <div
          css={{
            marginLeft: "auto",
            marginRight: "auto",
            [largeScreens]: {
              paddingTop: "80px"
            },
            [mediumScreens]: {
              paddingTop: "80px"
            }
          }}
        >
          <h3 css={{ [smallScreens]: { display: "none" } }}>Kontakt</h3>
          <div
            css={{
              paddingTop: "30px",
              listStyleType: "none",
              ">div": { paddingBottom: "18px" },
              " a": {
                textDecoration: "none",
                textAlign: "left",
                color: "inherit",
                fontWeight: "bold"
              },
              display: "grid",
              [smallScreens]: {
                marginLeft: "auto",
                marginRight: "auto",
                gridTemplateColumns: "1fr",
                maxWidth: "196px",
                paddingTop: "0px",
                paddingBottom: "60px"
              }
            }}
          >
            {socialLinks.map(({ icon, label, url, dataCy }) => (
              <div
                key={url}
                css={{
                  display: "grid",
                  gridTemplateColumns: "min-content auto",
                  ":hover": {
                    " a": {
                      color: "#ed253a",
                      ">div": {
                        backgroundColor: "#ed253a"
                      }
                    }
                  },
                  [smallScreens]: {
                    display: "block",
                    width: "40px",
                    height: "40px",
                    marginLeft: "auto",
                    marginRight: "auto"
                  }
                }}
              >
                <a aria-label={label} href={url} css={{}}>
                  <div
                    css={{
                      backgroundColor: "#e3e3e3",
                      mask: `url(${icon}) no-repeat center center`,
                      maskSize: "100% 100%",
                      width: "28px",
                      height: "28px",
                      marginRight: "16px",
                      marginTop: "-5px",
                      ":hover": {
                        backgroundColor: "#ed253a"
                      },
                      [smallScreens]: {
                        width: "40px",
                        height: "40px"
                      }
                    }}
                  />
                </a>
                <a
                  css={{ [smallScreens]: { display: "none" } }}
                  href={url}
                  data-cy={dataCy}
                >
                  {label}
                </a>
              </div>
            ))}
          </div>
        </div>
        <div
          css={{
            [largeScreens]: { display: "none" },
            [mediumScreens]: { display: "none" }
          }}
        >
          <p
            data-cy="footer-address"
            css={{ ...finePrint, marginBottom: "5px" }}
          >
            Amsel.tech | Amselweg 7, 73266 Bissingen an der Teck
          </p>
          <p
            data-cy="footer-dev-plug"
            css={{ ...finePrint, fontWeight: "400" }}
          >
            {"< />"} with ♡ by Amsel-tech
          </p>
          <p
            data-cy="footer-copyright-notice"
            css={{ ...finePrint, marginTop: "6px", paddingBottom: "32px" }}
          >
            Copyright © {moment().year()} Amsel.tech
          </p>
        </div>
      </div>
    </ContentWrapper>
  </footer>
);

export default Footer;
