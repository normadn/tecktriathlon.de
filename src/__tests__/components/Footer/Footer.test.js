import React from "react";
import Footer from "../../../components/Footer";
import { shallow } from "enzyme";

describe("Footer", () => {
  it("renders correctly", () => {
    expect(shallow(<Footer />)).toMatchSnapshot();
  });
});
