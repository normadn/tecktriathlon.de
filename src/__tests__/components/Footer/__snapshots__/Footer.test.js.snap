// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<footer
  css={
    Object {
      "backgroundColor": "#11213f",
      "color": "#ffffff",
    }
  }
>
  <_default>
    <div
      css={
        Object {
          " h3": Object {
            "color": "inherit",
            "textTransform": "uppercase",
          },
          "@media(max-width: 950px)": Object {
            "display": "block",
            "textAlign": "center",
          },
          "display": "grid",
          "gridTemplateColumns": "auto auto auto",
          "paddingTop": "80px",
        }
      }
    >
      <div
        css={
          Object {
            "@media(max-width: 950px)": Object {
              "marginLeft": "auto",
              "marginRight": "auto",
            },
            "maxWidth": "455px",
          }
        }
      >
        <span>
          <img
            alt="QHacks Crown"
            css={
              Object {
                "marginBottom": "5px",
                "paddingRight": "13px",
              }
            }
            height="34px"
            src="test-file-stub"
          />
          <img
            alt="QHacks Wordmark"
            height="37px"
            src="test-file-stub"
          />
        </span>
        <p
          css={
            Object {
              "@media(max-width: 950px)": Object {
                "paddingLeft": "24px",
                "paddingRight": "24px",
              },
              "fontSize": "12px",
              "lineHeight": "1.67",
              "paddingBottom": "40px",
              "paddingRight": "20px",
              "paddingTop": "40px",
            }
          }
        >
          QHacks is Queen’s University’s annual hackathon! QHacks was founded in 2016 with a mission to advocate and incubate the tech community at Queen’s University and throughout Canada. Going into our fourth year we have been aggressively growing to become one of the biggest and most exciting hackathons in North America.
        </p>
        <h3>
          Stay Connected
        </h3>
        <p
          css={
            Object {
              "fontSize": "12px",
              "lineHeight": "20px",
              "paddingBottom": "4px",
              "paddingTop": "12px",
            }
          }
        >
          Sign up for our mailing list for the latest updates.
        </p>
        <div
          css={
            Object {
              "@media(max-width: 950px)": Object {
                "marginLeft": "0px",
                "paddingBottom": "24px",
              },
              "alignItems": "left",
              "display": "flex",
              "justifyContent": "left",
              "marginLeft": "-8px",
              "paddingBottom": "64px",
            }
          }
        >
          <MediaQuery
            query="screen and (max-width: 950px)"
            values={Object {}}
          >
            <Apollo(SignUpForm)
              backgroundColor="#c81c2e"
              center={true}
              dataCy="footer-signup-button"
              fontSize={14}
              foregroundColor="white"
            />
          </MediaQuery>
          <MediaQuery
            query="screen and (min-width: 950px)"
            values={Object {}}
          >
            <Apollo(SignUpForm)
              backgroundColor="#c81c2e"
              dataCy="footer-signup-button"
              fontSize={14}
              foregroundColor="white"
            />
          </MediaQuery>
        </div>
        <div
          css={
            Object {
              "@media(max-width: 950px)": Object {
                "display": "none",
              },
            }
          }
        >
          <p
            css={
              Object {
                "color": "#bebebe",
                "fontSize": "12px",
                "marginBottom": "5px",
              }
            }
            data-cy="footer-address"
          >
            Queen's University | 99 University Ave, Kingston, ON
          </p>
          <a
            css={
              Object {
                "color": "#bebebe",
                "fontSize": "12px",
                "fontWeight": "400",
              }
            }
            data-cy="footer-dev-plug"
            href="https://github.com/qhacks/qhacks-website"
            rel="external noopener"
            target="_blank"
          >
            &lt; /&gt;
             with ♡ by Queen's students
          </a>
          <p
            css={
              Object {
                "color": "#bebebe",
                "fontSize": "12px",
                "marginTop": "6px",
                "paddingBottom": "32px",
              }
            }
            data-cy="footer-copyright-notice"
          >
            Copyright © 
            2019
             QHacks
          </p>
        </div>
      </div>
      <div
        css={
          Object {
            "@media(max-width: 950px)": Object {
              "paddingBottom": "50px",
              "paddingTop": "0px",
            },
            "marginLeft": "auto",
            "marginRight": "auto",
            "paddingTop": "80px",
          }
        }
      >
        <h3>
          Navigate
        </h3>
        <ul
          css={
            Object {
              " a": Object {
                ":hover": Object {
                  "color": "#ed253a",
                },
                "color": "inherit",
                "fontWeight": 600,
                "textDecoration": "none",
              },
              ">li": Object {
                "paddingBottom": "23px",
              },
              "listStyleType": "none",
              "paddingTop": "30px",
            }
          }
        >
          <li
            key="https://2018.qhacks.io"
          >
            <a
              data-cy="footer-2018-link"
              href="https://2018.qhacks.io"
            >
              QHacks 2018 ▸
            </a>
          </li>
          <li
            key="https://localhackday.mlh.io/"
          >
            <a
              data-cy="footer-lhd-link"
              href="https://localhackday.mlh.io/"
            >
              Local Hack Day ▸
            </a>
          </li>
          <li
            key="https://medium.com/@qhacks"
          >
            <a
              data-cy="footer-blog-link"
              href="https://medium.com/@qhacks"
            >
              QHacks Blog ▸
            </a>
          </li>
          <li
            key="https://app.qhacks.io"
          >
            <a
              data-cy="footer-dashboard-link"
              href="https://app.qhacks.io"
            >
              QHacks Dashboard ▸
            </a>
          </li>
          <li
            key="https://static.mlh.io/docs/mlh-code-of-conduct.pdf"
          >
            <a
              data-cy="footer-mlh-code-of-conduct-link"
              href="https://static.mlh.io/docs/mlh-code-of-conduct.pdf"
            >
              MLH Code of Conduct ▸
            </a>
          </li>
        </ul>
      </div>
      <div
        css={
          Object {
            "@media (min-width: 950px) and (max-width: 1200px)": Object {
              "paddingTop": "80px",
            },
            "@media(min-width: 1200px)": Object {
              "paddingTop": "80px",
            },
            "marginLeft": "auto",
            "marginRight": "auto",
          }
        }
      >
        <h3
          css={
            Object {
              "@media(max-width: 950px)": Object {
                "display": "none",
              },
            }
          }
        >
          Get in touch
        </h3>
        <div
          css={
            Object {
              " a": Object {
                "color": "inherit",
                "fontWeight": "bold",
                "textAlign": "left",
                "textDecoration": "none",
              },
              ">div": Object {
                "paddingBottom": "18px",
              },
              "@media(max-width: 950px)": Object {
                "gridTemplateColumns": "1fr 1fr 1fr 1fr",
                "marginLeft": "auto",
                "marginRight": "auto",
                "maxWidth": "196px",
                "paddingBottom": "60px",
                "paddingTop": "0px",
              },
              "display": "grid",
              "listStyleType": "none",
              "paddingTop": "30px",
            }
          }
        >
          <div
            css={
              Object {
                ":hover": Object {
                  " a": Object {
                    ">div": Object {
                      "backgroundColor": "#ed253a",
                    },
                    "color": "#ed253a",
                  },
                },
                "@media(max-width: 950px)": Object {
                  "display": "block",
                  "height": "40px",
                  "marginLeft": "auto",
                  "marginRight": "auto",
                  "width": "40px",
                },
                "display": "grid",
                "gridTemplateColumns": "min-content auto",
              }
            }
            key="mailto:hello@qhacks.io"
          >
            <a
              aria-label="hello@qhacks.io"
              css={Object {}}
              href="mailto:hello@qhacks.io"
            >
              <div
                css={
                  Object {
                    ":hover": Object {
                      "backgroundColor": "#ed253a",
                    },
                    "@media(max-width: 950px)": Object {
                      "height": "40px",
                      "width": "40px",
                    },
                    "backgroundColor": "#e3e3e3",
                    "height": "28px",
                    "marginRight": "16px",
                    "marginTop": "-5px",
                    "mask": "url(test-file-stub) no-repeat center center",
                    "maskSize": "100% 100%",
                    "width": "28px",
                  }
                }
              />
            </a>
            <a
              css={
                Object {
                  "@media(max-width: 950px)": Object {
                    "display": "none",
                  },
                }
              }
              data-cy="footer-email-link"
              href="mailto:hello@qhacks.io"
            >
              hello@qhacks.io
            </a>
          </div>
          <div
            css={
              Object {
                ":hover": Object {
                  " a": Object {
                    ">div": Object {
                      "backgroundColor": "#ed253a",
                    },
                    "color": "#ed253a",
                  },
                },
                "@media(max-width: 950px)": Object {
                  "display": "block",
                  "height": "40px",
                  "marginLeft": "auto",
                  "marginRight": "auto",
                  "width": "40px",
                },
                "display": "grid",
                "gridTemplateColumns": "min-content auto",
              }
            }
            key="https://twitter.com/qhacks19"
          >
            <a
              aria-label="Follow us on Twitter"
              css={Object {}}
              href="https://twitter.com/qhacks19"
            >
              <div
                css={
                  Object {
                    ":hover": Object {
                      "backgroundColor": "#ed253a",
                    },
                    "@media(max-width: 950px)": Object {
                      "height": "40px",
                      "width": "40px",
                    },
                    "backgroundColor": "#e3e3e3",
                    "height": "28px",
                    "marginRight": "16px",
                    "marginTop": "-5px",
                    "mask": "url(test-file-stub) no-repeat center center",
                    "maskSize": "100% 100%",
                    "width": "28px",
                  }
                }
              />
            </a>
            <a
              css={
                Object {
                  "@media(max-width: 950px)": Object {
                    "display": "none",
                  },
                }
              }
              data-cy="footer-twitter-link"
              href="https://twitter.com/qhacks19"
            >
              Follow us on Twitter
            </a>
          </div>
          <div
            css={
              Object {
                ":hover": Object {
                  " a": Object {
                    ">div": Object {
                      "backgroundColor": "#ed253a",
                    },
                    "color": "#ed253a",
                  },
                },
                "@media(max-width: 950px)": Object {
                  "display": "block",
                  "height": "40px",
                  "marginLeft": "auto",
                  "marginRight": "auto",
                  "width": "40px",
                },
                "display": "grid",
                "gridTemplateColumns": "min-content auto",
              }
            }
            key="https://fb.com/qhacks"
          >
            <a
              aria-label="Like us on Facebook"
              css={Object {}}
              href="https://fb.com/qhacks"
            >
              <div
                css={
                  Object {
                    ":hover": Object {
                      "backgroundColor": "#ed253a",
                    },
                    "@media(max-width: 950px)": Object {
                      "height": "40px",
                      "width": "40px",
                    },
                    "backgroundColor": "#e3e3e3",
                    "height": "28px",
                    "marginRight": "16px",
                    "marginTop": "-5px",
                    "mask": "url(test-file-stub) no-repeat center center",
                    "maskSize": "100% 100%",
                    "width": "28px",
                  }
                }
              />
            </a>
            <a
              css={
                Object {
                  "@media(max-width: 950px)": Object {
                    "display": "none",
                  },
                }
              }
              data-cy="footer-facebook-link"
              href="https://fb.com/qhacks"
            >
              Like us on Facebook
            </a>
          </div>
          <div
            css={
              Object {
                ":hover": Object {
                  " a": Object {
                    ">div": Object {
                      "backgroundColor": "#ed253a",
                    },
                    "color": "#ed253a",
                  },
                },
                "@media(max-width: 950px)": Object {
                  "display": "block",
                  "height": "40px",
                  "marginLeft": "auto",
                  "marginRight": "auto",
                  "width": "40px",
                },
                "display": "grid",
                "gridTemplateColumns": "min-content auto",
              }
            }
            key="https://instagram.com/qhacks19"
          >
            <a
              aria-label="Follow us on Instagram"
              css={Object {}}
              href="https://instagram.com/qhacks19"
            >
              <div
                css={
                  Object {
                    ":hover": Object {
                      "backgroundColor": "#ed253a",
                    },
                    "@media(max-width: 950px)": Object {
                      "height": "40px",
                      "width": "40px",
                    },
                    "backgroundColor": "#e3e3e3",
                    "height": "28px",
                    "marginRight": "16px",
                    "marginTop": "-5px",
                    "mask": "url(test-file-stub) no-repeat center center",
                    "maskSize": "100% 100%",
                    "width": "28px",
                  }
                }
              />
            </a>
            <a
              css={
                Object {
                  "@media(max-width: 950px)": Object {
                    "display": "none",
                  },
                }
              }
              data-cy="footer-instagram-link"
              href="https://instagram.com/qhacks19"
            >
              Follow us on Instagram
            </a>
          </div>
        </div>
      </div>
      <div
        css={
          Object {
            "@media (min-width: 950px) and (max-width: 1200px)": Object {
              "display": "none",
            },
            "@media(min-width: 1200px)": Object {
              "display": "none",
            },
          }
        }
      >
        <p
          css={
            Object {
              "color": "#bebebe",
              "fontSize": "12px",
              "marginBottom": "5px",
            }
          }
          data-cy="footer-address"
        >
          Queen's University | 99 University Ave, Kingston, ON
        </p>
        <a
          css={
            Object {
              "color": "#bebebe",
              "fontSize": "12px",
              "fontWeight": "400",
            }
          }
          data-cy="footer-dev-plug"
          href="https://github.com/qhacks/qhacks-website"
          rel="external noopener"
          target="_blank"
        >
          &lt; /&gt;
           with ♡ by Queen's students
        </a>
        <p
          css={
            Object {
              "color": "#bebebe",
              "fontSize": "12px",
              "marginTop": "6px",
              "paddingBottom": "32px",
            }
          }
          data-cy="footer-copyright-notice"
        >
          Copyright © 
          2019
           QHacks
        </p>
      </div>
    </div>
  </_default>
</footer>
`;
